internal/network/manifests_generator.go

```diff
@@ -68,6 +69,81 @@ spec:
   osImageURL: ""
 `
 
+const snoDnsmasqConf = `
+address=/{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}/{{.HOST_IP}}
+addn-hosts=/etc/api-int.host
```

Comment:
Are you ever planning to change the api-int record after initial deployment? If not, you can just make this another address entry and skip the separate file. The only reason it was separate in my work was that we needed the operator to be able to change it and didn't want to re-template the entire config file.